posthog.go

```diff
@@ -711,13 +711,21 @@ func (c *client) getFeatureVariantsWithOptions(distinctId string, groups Groups,
 }
 
 func (c *client) makeRemoteConfigRequest(flagKey string) (string, error) {
-	remoteConfigEndpoint := fmt.Sprintf("api/projects/@current/feature_flags/%s/remote_config/", flagKey)
-	url, err := url.Parse(c.Endpoint + "/" + remoteConfigEndpoint)
+	baseURL, err := url.JoinPath(c.Endpoint, "api/projects/@current/feature_flags", flagKey, "remote_config")
```

Comment:
url.JoinPath is probably available only in some new versions of the std library, 

In general, posthog-go is tested against some ancient versions of Go, I've checked and most companies (including Go) support ~2 previous versions.

The Go states:
>Each major Go release is supported until there are two newer major releases. For example, Go 1.5 was supported until the Go 1.7 release, and Go 1.6 was supported until the Go 1.8 release. We fix critical problems, including [critical security problems](https://go.dev/security), in supported releases as needed by issuing minor revisions (for example, Go 1.6.1, Go 1.6.2, and so on).

and I think we should do something similar (or maybe 1/2 more version or so)

Reply:
Although I'm reading that 1.21 has enhanced back compat features. I think I'll go with your recommendation @orian and move us to `1.21`.

Reply:
<3 sounds great to me